### ISSUE TITLE
YAML Front matter does not display if line endings are Windows-style (\r\n)

### ISSUE THREAD

**Issue by miggland** [gollum/gollum#2066]
**Describe the bug**

This bug has some similarities to this issue: https://github.com/gollum/gollum/issues/1617

When files are created on Windows, with Windows-styled line endings `\r\n`, frontmatter is not read properly. Reading via macros (using `gollum_page.metadata`) does not work either.
If opening in the Gollum editor and saving, the same YAML is parsed properly.

The rest of the markdown file seems unaffected.

**To Reproduce**
1. Create a file on windows, give it some front matter. Save in a file with \r\n line endings
[testfile_rn.md](https://github.com/user-attachments/files/16710531/testfile_rn.md)
2. View in Gollum. See this:
![image](https://github.com/user-attachments/assets/21e67ff6-e3c0-465d-98bd-d0c7d7edca7a)
3. Change the line-endings in an editor (eg Geany), or in Gollum: Edit -> Save (without making any manual changes)
The new file has line endings \n:
[testfile_n.md](https://github.com/user-attachments/files/16710549/testfile_n.md)

4. Go back to view the page. See this:
![image](https://github.com/user-attachments/assets/090b901b-8501-492f-b3ad-3efc33082fc7)

**Expected behavior**
The YAML frontmatter should be read even on files with \r\n endings.

**Environment Info**
Docker, image `gollumwiki/gollum:6.0`.

```
$ docker compose -f gollum-isodocs.yml exec wiki gollum --version
Gollum 6.0.1

```

**Comment by bartkamphorst** [comment]
Can confirm this issue on master.

**Comment by bartkamphorst** [comment]
I narrowed down the issue to calls to `sanitize` in the yaml filter in `gollum-lib`. With the supplied test case, the captured string that was being passed to `sanitize` looked like this:
```ruby
"---\r\ntitle: Test document\r\nuse: documentation\r\ntags:\r\n - important\r\n - concise\r\n"
```
Sanitize would turn this into:
```ruby
"---&#13;\ntitle: Test document&#13;\nuse: documentation&#13;\ntags:&#13;\n - important&#13;\n - concise&#1
3;\n"
```
Which would trigger the `Psych::SyntaxError` in the `::YAML:safe_load` call.
```
(<unknown>): mapping values are not allowed in this context at line 2 column 6 (Psych::SyntaxError)
```

The good news, however, is that the issue no longer occurs with the latest `gollum-lib` after @dometto [refactored](https://github.com/gollum/gollum-lib/blob/master/lib/gollum-lib/filter/yaml.rb) the `sanitize_frontmatter` code.

This tracks, as `sanitize` is now called on the individual elements of the frontmatter instead of on the full string. In other words, the YAML is loaded before sanitize is called, and YAML knows how to handle Windows line endings:
```ruby
yml =  "---\r\ntitle: Test document\r\nuse: documentation\r\ntags:\r\n - important\r\n - concise\r\n"
 => "---\r\ntitle: Test document\r\nuse: documentation\r\ntags:\r\n - important\r\n - concise\r\n" 
3.1.2 :003 > puts ::YAML.safe_load(yml)
{"title"=>"Test document", "use"=>"documentation", "tags"=>["important", "concise"]}
```

So the recommendation is to update your install to use the latest gollum-lib, and all should be fine. 👍

**Comment by bartkamphorst** [comment]
Actually, maybe `gollum-lib` hasn't been released with this fix yet? @dometto ?

**Comment by miggland** [comment]
Hi. Thanks for looking into this.

I've updated to `Gollum 6.1.0`, container tag 6.1, but the error still appears in the same way, presumably because `gollum-lib` is still at version 6.0..

**Comment by bartkamphorst** [comment]
Indeed. This should be fixed with the next bug-fix release of gollum-lib.
